docs/src/test-global-setup-teardown-js.md

```diff
@@ -26,25 +45,181 @@ export default defineConfig({
+
+In this example the 'logged in chromium' project depends on the setup project whereas the 'logged out chromium' project does not depend on the setup project, and does not use the `storageState` option.
+
+```js title="playwright.config.ts"
+import { defineConfig } from '@playwright/test';
+
+export const STORAGE_STATE = path.join(__dirname, 'playwright/.auth/user.json');
+
+export default defineConfig({
+  use: {
+    baseURL: 'http://localhost:3000/',
+  },
+  projects: [
+    {
+      name: 'setup',
+      testMatch: /global.setup\.ts/,
     },
     {
-      name: 'webkit',
-      use: { ...devices['Desktop Safari'] },
+      name: 'logged in chromium',
+      use: { ...devices['Desktop Chrome'] },
+      testMatch: '**/*.loggedin.spec.ts',
       dependencies: ['setup'],
+      use: {
+        storageState: STORAGE_STATE,
+      },
+    },
+    {
+      name: 'logged out chromium',
+      use: { ...devices['Desktop Chrome'] },
+      testIgnore: ['**/*loggedin.spec.ts']
     },
   ],
 });
+```
+
+We then create a setup test, stored at root level of your project, that logs in to an application and populates the context with the storage state after the login actions have been performed. By doing this you only have to log in once and the credentials will be stored in the `STORAGE_STATE` file, meaning you don't need to log in again for every test. Start by importing the `STORAGE_STATE` from the Playwright config file and then use this as the path to save your storage state to the page's context.
+
+```js title="global.setup.ts"
+import { test as setup, expect } from '@playwright/test';
+import { STORAGE_STATE } from '../playwright.config';
+
+setup('do login', async ({ page }) => {
+  await page.goto('/');
+  await page.getByLabel('User Name').fill('user');
+  await page.getByLabel('Password').fill('password');
+  await page.getByText('Sign in').click();
+
+  await page.context().storageState({ path: STORAGE_STATE });
+});
+```
+
+```js title="tests/menu.loggedin.spec.ts"
+import { test, expect } from '@playwright/test';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/');
+});
+
+test('menu', async ({ page }) => {
+  // You are signed in!
+})
+```
+
+For a more detailed example check out our blog post: [A better global setup in Playwright reusing login with project dependencies](https://dev.to/playwright/a-better-global-setup-in-playwright-reusing-login-with-project-dependencies-14) or check the [v1.31 release video](https://youtu.be/PI50YAPTAs4) to see the demo.
+
+### Teardown
+
+You can teardown your setup by adding a [`property: TestProject.teardown`] property to your setup project. This will run after all dependencies have run.
```

Comment:
```suggestion
You can teardown your setup by adding a [`property: TestProject.teardown`] property to your setup project. This will run after all dependent projects have run.
```